engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -240,16 +240,10 @@ private ConstrainedField findPropertyMetaData(Field field) {
 		);
 
 		boolean isCascading = field.isAnnotationPresent( Valid.class );
-		UnwrapValidatedValue unwrapValidatedValue = field.getAnnotation( UnwrapValidatedValue.class );
-		UnwrapMode unwrapMode = UnwrapMode.AUTOMATIC;
-		if ( unwrapValidatedValue != null ) {
-			unwrapMode = unwrapValidatedValue.value() ? UnwrapMode.UNWRAP : UnwrapMode.SKIP_UNWRAP;
-		}
-
 		Set<MetaConstraint<?>> typeArgumentsConstraints = findTypeAnnotationConstraintsForMember( field );
-		if ( !typeArgumentsConstraints.isEmpty() && !ReflectionHelper.isIterable( ReflectionHelper.typeOf( field ) ) ) {
-			unwrapMode = UnwrapMode.UNWRAP;
-		}
+
+		boolean typeArgumentAnnotated = !typeArgumentsConstraints.isEmpty();
+		UnwrapMode unwrapMode = unwrapMode( field, typeArgumentAnnotated );
```

Comment:
true, true - I am not arguing against this. I think we even had the same discussion when Khalid was working on this. In the end we would need to refactor the metadata a bit. AFAICR we are using the field metadata to host this configuration value, because the annotation is hosted on the field (property). 

I am all for re-visiting the metadata. There are several things which could/should be improved. Just not sure whether it is necessary right now. At least not as part of this issue I think.

Does this make sense?